src/main/java/org/eclipsefoundation/marketplace/resource/InstallResource.java

```diff
@@ -121,21 +124,39 @@ public Response selectInstallMetrics(@PathParam("listingId") String listingId, @
 	@Path("/{listingId}/{version}")
 	public Response postInstallMetrics(@PathParam("listingId") String listingId, @PathParam("version") String version,
 			Install installDetails) {
-		// update the install details to reflect the current request
-		installDetails.setInstallDate(new Date(System.currentTimeMillis()));
-		installDetails.setListingId(listingId);
-		installDetails.setVersion(version);
+		Install record = null;
 		
+		// check that connection was opened by MPC, and check for install information
```

Comment:
I don't get why we rely on user agent to get such the "install record". They should be passed via query parameters, or custom headers. Relying on a specific string format for the user agent is flaky, not testable, and not future proof / extensible.

Reply:
I agree that it isn't the most robust solution. This was captured as a wanted addition to the API as an option for posting. That's why I wrapped additional logic to capture the JSON body, and check that the agent reads as from MPC before allowing the install, as they are the only consumer of this endpoint outside of our initial set up (where we can spoof the agent). This way, should the MPC agent change, there is a backup of reading the JSON.

If you want to look at removing this, which I wouldn't be very opposed to, we should talk w/ Chris + potentially the guys from MPC to see how important this feature is.

Reply:
@chrisguindon This stuff is really just hidden API. The current implementation looks ok to me so I'll +1 this PR, but IMO you should keep in mind to make this first class API.